### PR TITLE
Implement whitelist for limiting bid values

### PR DESCRIPTION
**Pull request by loredanacirstea**

### CODE REVIEW

File: contracts/auction.sol [hunk 1]
```diff
@@ -213,6 +231,7 @@ contract DutchAuction {
         atStage(Stages.AuctionStarted)
     {
         require(msg.value > 0);
+        require(bids[msg.sender] + msg.value < bid_threshold || whitelist[msg.sender]);
```
Comment (ulope) [inline contracts/auction.sol]: From the comment at the top `Bid value over which` this should this be `<=`.

Comment (loredanacirstea) [reply contracts/auction.sol]: Fixed. Thank you.

File: contracts/auction.sol [hunk 2]
```diff
@@ -173,6 +181,16 @@ contract DutchAuction {
         price_exponent = _price_exponent;
     }
 
+    /// @notice Adds account addresses to whitelist.
+    /// @dev Adds account addresses to whitelist.
+    /// @param _bidder_addresses Array of addresses.
+    function addToWhitelist(address[] _bidder_addresses) public isOwner {
+        for (uint32 i = 0; i < _bidder_addresses.length; i++) {
+            assert(_bidder_addresses[i] != 0x0);
```
Comment (LefterisJP) [inline contracts/auction.sol]: Why do we need this? Couldn't the whitelist in theory be also populated before auction starts?

Comment (loredanacirstea) [reply contracts/auction.sol]: I don't understand - why do we need the `assert`?
The whitelist can be populated befor auction starts - I don't have a stage requirement.

Comment (LefterisJP) [reply contracts/auction.sol]: I believe we should be able to start populating the whitelist before auction starts.

Comment (loredanacirstea) [reply contracts/auction.sol]: Removed that assert anyway, as per your proposal. Worst case scenario, we whitelist `0x0`

**Pull request merged**